tests/jballerina-unit-test/src/test/resources/test-src/annotations/local_record_annotation_test.bal

```diff
@@ -0,0 +1,181 @@
+                                    assertEquality("typedesc map", (typeof m).toString());
+                                    assertEquality("{\"annot\":{\"value\":\"chiranS\"}}", m.toString()); 
+                                };
+                                return func;
+                                };
+    function() func = var1();
+    func();
+}
+
+function testAnnotationOnLocalRecordWithinNestedLambdaFunctions1() {
+    string k = "chiranS";
+    function (string) returns function() var1 = function (string val) returns function() {
+                                function() func = function () {
+                                    record {@annot {value: val} string x;} r = {x : ""}; 
+                                    any m = getLocalRecordAnnotations(typeof r, "$field$.x");
+                                    assertEquality("typedesc map", (typeof m).toString());
+                                    assertEquality("{\"annot\":{\"value\":\"chiranS\"}}", m.toString()); 
+                                };
+                                return func;
+                                };
+    function() func = var1(k);
+    func();
+}
+
+function testAnnotationOnLocalRecordWithinNestedLambdaFunctions2() {
+    string k = "chiranS";
+    function (string) returns function() var1 = function (string val) returns function() {
+                                string name = "Name: ";
+                                function() func = function () {
+                                    record {@annot {value: name + val} string x;} r = {x : ""}; 
+                                    any m = getLocalRecordAnnotations(typeof r, "$field$.x");
+                                    assertEquality("typedesc map", (typeof m).toString());
+                                    assertEquality("{\"annot\":{\"value\":\"Name: chiranS\"}}", m.toString()); 
+                                };
+                                return func;
+                                };
+    function() func = var1(k);
+    func();
+}
+
+function testMultipleAnnotationsOnLocaRecord() {
+    string k = "chiran";
+    record {@annot {value: k} @annot1 {value: k} string x;} r = {x : ""}; 
+
+    any m = getLocalRecordAnnotations(typeof r, "$field$.x");
+    assertEquality("typedesc map", (typeof m).toString());
+}
+
+function testLocalRecordAnnotations() {
+        testMultipleAnnotationsOnLocaRecord();
+        testAnnotationOnLocalRecord();
+        testAnnotationOnLocalRecordWithGlobalVariable();
+        testAnnotationOnLocalRecordWithinLambdaFunction();
+        testAnnotationOnLocalRecordWithinLambdaFunction1();
+        testAnnotationOnLocalRecordWithinNestedLambdaFunctions();
+        testAnnotationOnLocalRecordWithinNestedLambdaFunctions1();
+        testAnnotationOnLocalRecordWithinNestedLambdaFunctions2();
+        testAnnotationWithMultipleFieldsOnLocalRecord();
+        testAnnotationOnLocalRecordWithMultipleFields();
+}
```

Comment:
We usually use data providers on the Java side instead. That way the remaining tests will run even if one in the middle fails.